### PR TITLE
Fix a bug with computing the output mask after generate

### PR DESCRIPTION
**Pull request by mattdangerw**
We were calling `cumsum` with the wrong axis, meaning we were not correctly masking all positions after an end token.

### CODE REVIEW

File: keras_nlp/models/gpt2/gpt2_causal_lm.py [hunk 1]
```diff
@@ -253,6 +253,7 @@ def call_with_cache(
             self.backbone.get_layer("token_embedding").embeddings,
             transpose_b=True,
         )
+        print(logits)
```
Comment (chenmoneygithub) [inline keras_nlp/models/gpt2/gpt2_causal_lm.py:256]: delete this?

File: keras_nlp/models/gpt2/gpt2_causal_lm_test.py [hunk 2]
```diff
@@ -118,6 +120,23 @@ def test_generate(self):
             self.preprocessed_batch["padding_mask"][:, :5],
         )
 
+    def test_early_stopping(self):
```
Comment (chenmoneygithub) [inline keras_nlp/models/gpt2/gpt2_causal_lm_test.py:123]: Thanks for the test case, this is very cool

**Review by chenmoneygithub — APPROVED** [review]
Thanks! I am surprised we did not catch this....
